### PR TITLE
Design and add a MongoDB schema (Issue 75)

### PR DESCRIPTION
**Pull request by Kannav02**
This PR will add the MongoDB schema to support the feedback data that was being used with google sheets

The following additions were made 

- [x] function allowing to connect to the MongoDB client were established
- [x] new `.env.example` file was made to provide an example of making MongoDB URI
- [x] functions allowing to manipulate the MongoDB instance have been made 

Apart from this a minor bug fix was also done in the `frontend/utils/feedback.py` file, which allowed the right passing of parameters to the functions during the runtime to avoid unexpected errors

An example of when I tried to create a mongoDB instance is here , I got the collections and the database that were being created
Screenshot 2024-11-18 at 9 16 52 PM

This PR is one of many which would lead to a complete solution of issue #75

### CODE REVIEW

**Comment by luarss** [comment]
Please signoff your commits @Kannav02. If you are successful, the `Verified` logo will appear beside your commits. 

Example: 
image

File: common/mongoClient.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+load_dotenv()
+
+# HARDCODED collection names
+feedback_collection_name = "feedback"
+context_collection_name = "context"
+
+print(os.getenv("MONGO_DB_URI"))
+
+
+def get_mongo_client() -> Database:
```
Comment (luarss) [inline common/mongoClient.py:15]: Returns Database or MongoClient?

File: frontend/utils/feedback.py [hunk 2]
```diff
@@ -170,8 +171,15 @@ def show_feedback_form(
     feedback = st.sidebar.text_area("Please provide your feedback or report an issue:")
 
     if selected_question:
-        sources = [metadata[selected_question].get("sources", "N/A")]
-        context = [metadata[selected_question].get("context", "N/A")]
+        sources = metadata[selected_question].get("sources", ["N/A"])
```
Comment (luarss) [inline frontend/utils/feedback.py:174]: Can you explain this? Is it supposed to be a list of lists?

File: common/mongoClient.py (hunk 1 above)
Comment (Kannav02) [reply common/mongoClient.py:15]: ah , So when i was first making the design, I wanted it to return the mongoClient, but later on it was rarely used apart from just getting the database client instance, so I changed it, I'll change the semantics and fix this

File: common/mongoClient.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    - None
+    """
+    try:
+        client_database.create_collection(collection_name,validator=validator)
+        print("Collection created successfully")
+    except Exception as e:
+        print(f"Failed to create collection: {e}")
+        return None
+    
+submit_feedback()
```
[the remark this answers is not in the document]
Comment (Kannav02) [reply common/mongoClient.py:153]: this was just to see a demo run for the collections and db, I'll fix this as well

File: frontend/utils/feedback.py (hunk 2 above)
Comment (Kannav02) [reply frontend/utils/feedback.py:174]: so the `submit_feedback_to_google_sheet` function was returning an error even when the input was right, apparently what was causing this issue was the source and context was being passed in as a `str` but the function accepted a list, so I changed it to have a type-safety that whenever an independent string is being passed, it is changed to a list of single string so the function works

**Comment by Kannav02** [comment]
Hey @luarss , i believe this should be good for review now, do let me know if you want any other changes to this

thank you

**Comment by luarss** [comment]
Hi, can you please recreate the branch and PR directly from upstream and not the fork? I think there are some secrets not propagatable to forks.

**Comment by Kannav02** [comment]
I am trying to push my branch to upstream, but it says I don't have the permission to do so, I believe this is because of the write access 

` remote: Permission to The-OpenROAD-Project/ORAssistant.git denied to Kannav02.
fatal: unable to access 'https://github.com/The-OpenROAD-Project/ORAssistant.git/': The requested URL returned error: 403`

**Comment by luarss** [comment]
Do try again, given you access.

**Comment by Kannav02** [comment]
I believe you were right about the secrets not being able to propagate to the fork, this is quite interesting, I can look into this as well why it is happening if you want me to, as this might lead to each contributor needing to have write access to the repo before they can merge their changes

**Pull request closed without merging**